### PR TITLE
III-6266 Add new clusters for duplicate places

### PR DESCRIPTION
**Pull request by grubolsch**
### Added
Add new clusters in the duplicate_places_import table to the duplicate_places table, with a canonical as NULL.
These will be later picked up to be processed by another script.

### Changed
The "getClusterIds" function is changed to "getClusterIdsWithoutCanonical()" only return places that do not have a canonical in their cluster.

---

Ticket: https://jira.uitdatabank.be/browse/III-6266

### CODE REVIEW

File: src/Place/Canonical/DBALDuplicatePlaceRepository.php [hunk 1]
```diff
@@ -95,15 +97,39 @@ public function calculateNoLongerInCluster(): array
         return $this->processRawToClusterRecord($statement->fetchAllAssociative());
     }
 
-    public function addToDuplicatePlacesRemovedFromCluster(string $clusterId): void
+    /** @return ClusterRecord[] */
+    public function calculateNotYetInCluster(): array
     {
-        $this->connection->executeQuery('INSERT INTO duplicate_places_removed_from_cluster SET cluster_id  = :cluster_id', [':cluster_id' => $clusterId]);
+        $statement = $this->connection->executeQuery('
```
Comment (LucWollants) [inline src/Place/Canonical/DBALDuplicatePlaceRepository.php:103]: Use a query builder here

File: src/Place/Canonical/DBALDuplicatePlaceRepository.php [hunk 2]
```diff
@@ -95,15 +97,39 @@ public function calculateNoLongerInCluster(): array
...
 
+    /** @return ClusterRecord[] */
     private function processRawToClusterRecord(array $data): array
     {
         return array_map(function ($row): ClusterRecord {
             return ClusterRecord::fromArray($row);
         }, $data);
     }
+
+    public function addToDuplicatePlacesRemovedFromCluster(string $clusterId): void
```
Comment (LucWollants) [inline src/Place/Canonical/DBALDuplicatePlaceRepository.php:123]: Move to the new repo for `places_removed_from_cluster` table

File: src/Place/Canonical/DBALDuplicatePlaceRepository.php [hunk 3]
```diff
@@ -95,15 +97,39 @@ public function calculateNoLongerInCluster(): array
...
+    {
+        $this->connection->executeQuery('INSERT INTO duplicate_places_removed_from_cluster SET cluster_id  = :cluster_id', [':cluster_id' => $clusterId]);
+    }
+
+    public function addToDuplicatePlaces(string $clusterId, string $placeUuid, string $canonical = null): void
+    {
+        $this->connection->executeQuery(
```
Comment (LucWollants) [inline src/Place/Canonical/DBALDuplicatePlaceRepository.php:130]: Use querybuilder

File: src/Place/Canonical/DBALDuplicatePlaceRepository.php [hunk 4]
```diff
@@ -95,15 +97,39 @@ public function calculateNoLongerInCluster(): array
         return $this->processRawToClusterRecord($statement->fetchAllAssociative());
     }
 
-    public function addToDuplicatePlacesRemovedFromCluster(string $clusterId): void
+    /** @return ClusterRecord[] */
+    public function calculateNotYetInCluster(): array
```
Comment (LucWollants) [inline src/Place/Canonical/DBALDuplicatePlaceRepository.php:101]: This method name is using an action that will happen later in its name. This is just a method to get the new clusters.

**Comment by grubolsch** [comment]
All feedback has been processed

File: src/Place/DuplicatePlace/ImportDuplicatePlacesProcessor.php
Comment (LucWollants) [inline src/Place/DuplicatePlace/ImportDuplicatePlacesProcessor.php:1]: This seems to be an empty file?

File: src/Place/Canonical/ClusterRecordRow.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Place\Canonical;
+
+class ClusterRecordRow
```
Comment (LucWollants) [inline src/Place/Canonical/ClusterRecordRow.php:7]: Maybe this is a `PlaceWithClusterRecord`? Because a row is always unique on place id.

**Review by simon-debruijn — APPROVED** [review]

**Pull request merged**